### PR TITLE
filtering code added

### PR DESCRIPTION
**Pull request by naqvia**
This PR will add filtering (gnomad variable, gnomad af, and variant depth) to the workflow

### CODE REVIEW

**Comment by naqvia** [comment]
So I have added variant depth and gnomad filters, so that they are removed from downstream processing. It first goes through the original vcf file and removes all such variants. And then as the tool is ran it filters for variants in the filtered vcf file. I have defaults ... maybe we can decide on those/modify them. The command I ran was the following: 

`Rscript 01-annotate_variants_CAVATICA_input.R --vcf input/test-cavatica.vcf --multianno input/test-cavatica.hg38_multianno.txt --intervar input/test-cavatica.hg38_multianno.txt.intervar --autopvs1 input/test-cavatica.autopvs1.tsv --output out --variant_summary input/variant_summary.txt --submission_summary input/submission_summary.txt --gnomad_variable gnomad_3_1_1_AF_non_cancer --variant_depth 10`

File: AutoGVP/01-annotate_variants_CAVATICA_input.R [hunk 1]
```diff
@@ -56,7 +56,7 @@ option_list <- list(
               help = "variant_summary file (format: variant_summary_2023-02.txt)"),
   make_option(c("--submission_summary"), type = "character",
               help = "specific submission summary file (format: submission_summary.txt)"),
-  make_option(c("--gnomad_variable"), type = "character",default = "Freq_gnomAD_genome_ALL",
+  make_option(c("--gnomad_variable"), type = "character",default = "gnomad_3_1_1_AF_non_cancer",
```
Comment (jharenza) [inline AutoGVP/01-annotate_variants_CAVATICA_input.R:59]: I don't think all files will have this default column - @jungkim2 can you provide all of the default column annots with VEP (ie. - is gnomad any version always run, is 1000g always run) - for this column, i think @migbro specifically named it this way, so it will not pertain to user input

Comment (jharenza) [reply AutoGVP/01-annotate_variants_CAVATICA_input.R:59]: probably also better to name this `--pop_freq_variable`

File: AutoGVP/01-annotate_variants_CAVATICA_input.R [hunk 2]
```diff
@@ -97,13 +98,16 @@ Sys.setenv("VROOM_CONNECTION_SIZE" = 131072 * 2)
...
   clinvar_vcf <- clinvar_vcf %>% 
                   dplyr::mutate(
-                    variant_depth = if_else( as.integer( str_match(INFO, "DP\\=(\\d+)")[, 2])  > filter_variant_depth, "PASS","FAIL"),
-                    gnomad_af     = if_else( as.numeric( str_match(INFO, gnomad_var_to_match)[,2])  > filter_variant_af, "PASS","FAIL"),
-                    variant_af    = if_else(as.integer(str_match(Sample, ":(\\d+)\\,(\\d+)") [,3]) / ( (as.integer(str_match(Sample, ":(\\d+)\\,(\\d+)") [,2]) ) + as.integer(str_match(Sample, ":(\\d+)\\,(\\d+)") [,3] )) > filter_variant_af, "PASS", "FAIL")
+                    variant_depth = as.integer( str_match(INFO, "DP\\=(\\d+)")[, 2]),
+                    gnomad_af     = as.numeric( str_match(INFO, gnomad_var_to_match)[,2]),
+                    #variant_af    = as.integer(str_match(Sample, ":(\\d+)\\,(\\d+)") [,3]) / ( (as.integer(str_match(Sample, ":(\\d+)\\,(\\d+)") [,2]) ) + as.integer(str_match(Sample, ":(\\d+)\\,(\\d+)") [,3] )) 
                     )
+  clinvar_vcf <- clinvar_vcf %>% dplyr::filter(variant_depth > filter_variant_depth & gnomad_af > filter_gnomad_af)
```
Comment (jharenza) [inline AutoGVP/01-annotate_variants_CAVATICA_input.R:109]:
```suggestion
  clinvar_vcf <- clinvar_vcf %>% 
  dplyr::filter(variant_depth >= filter_variant_depth & gnomad_af >= filter_gnomad_af) 
```
can you move the filter to the second line for readability, and I think we may also want to use this as min depth/AF

File: AutoGVP/01-annotate_variants_CAVATICA_input.R [hunk 3]
```diff
@@ -161,11 +165,12 @@ address_ambiguous_calls <- function(results_tab_abridged)
 }
 
 ## retrieve and store clinVar input file into table data.table::fread()
-vcf_input <-  vroom(input_clinVar_file, comment = "#",delim="\t", col_names = c("CHROM","START","ID","REF","ALT","QUAL","FILTER","INFO","FORMAT","Sample"), show_col_types = TRUE)
+vcf_input <-  vroom("/Users/naqvia/Documents/GitHub/pathogenicity-assessment/AutoGVP/input/test-cavatica.vcf", comment = "#",delim="\t", col_names = c("CHROM","START","ID","REF","ALT","QUAL","FILTER","INFO","FORMAT","Sample"), show_col_types = TRUE)
```
Comment (jharenza) [inline AutoGVP/01-annotate_variants_CAVATICA_input.R:168]: need to change this path

File: AutoGVP/01-annotate_variants_CAVATICA_input.R [hunk 4]
```diff
@@ -233,7 +230,7 @@ additional_intervar_cases <-  filter(clinvar_anno_vcf_df, final_call!="Benign",f
   
 ## filter only those variant entries that need an InterVar run (No Star) and add the additional intervar cases from above
 entries_for_intervar <- filter(clinvar_anno_vcf_df, Stars == "0", na.rm = TRUE) %>% 
-  bind_rows((additional_intervar_cases)) #%>% bind_rows(clinvar_anti_join_vcf_df)
+  bind_rows((additional_intervar_cases))
```
Comment (jharenza) [inline AutoGVP/01-annotate_variants_CAVATICA_input.R:233]:
```suggestion
  bind_rows(additional_intervar_cases) 
```

File: AutoGVP/01-annotate_variants_CAVATICA_input.R [hunk 5]
```diff
@@ -253,7 +249,7 @@ multianno_df  <-  vroom(input_multianno_file, delim="\t",trim_ws = TRUE, col_nam
   group_by(vcf_id) %>%
   arrange(vcf_id) %>%
   filter(row_number()==1) %>% 
-  ungroup
+  ungroup
```
Comment (jharenza) [inline AutoGVP/01-annotate_variants_CAVATICA_input.R:252]:
```suggestion
  ungroup()
```

File: AutoGVP/01-annotate_variants_CAVATICA_input.R [hunk 6]
```diff
@@ -84,9 +84,10 @@ summary_level <- opt$summary_level_vcf
 output_name   <- opt$output
 
 ## filters for gnomAD
-filter_gnomad_var    <- opt$gnomad_variable
+gnomad_variable    <- opt$gnomad_variable
 filter_variant_depth <- opt$variant_depth
 filter_variant_af    <- opt$variant_af
+filter_gnomad_af    <- opt$gnomad_af
```
Comment (jharenza) [inline AutoGVP/01-annotate_variants_CAVATICA_input.R:90]: I don't think you need these chunks actually - you can use `opt$XXX` within the code below

File: AutoGVP/01-annotate_variants_CAVATICA_input.R (hunk 1 above)
Comment (jungkim2) [reply AutoGVP/01-annotate_variants_CAVATICA_input.R:59]: For VEP annotation, I don't necessarily annotate with gnomAd and/or 1000g and --gnomad_variable filtering is not required. So it would be up to each user to be annotated right?

Comment (jungkim2) [reply AutoGVP/01-annotate_variants_CAVATICA_input.R:59]: I agree with the naming --pop_freq_variable as it could be gnomad/1000g/ESP/etc

Comment (jharenza) [reply AutoGVP/01-annotate_variants_CAVATICA_input.R:59]: ok gotcha - yes, we were seeing run times on smaller files of 3 hours on ec2 with no filtering, but maybe we need to customize our EC2 template, and for this @naqvia we should leave this as optional - no default

Comment (naqvia) [reply AutoGVP/01-annotate_variants_CAVATICA_input.R:59]: I think in order to make this usable and universal. We are observing that without filtering, we are running into long run times and memory-related crashes (cc @rjcorb). So, I think if we do not require a filter, we need at least to add a disclaimer or a warning for the end-user as the tool will likely crash and/or take a long time to run.

File: AutoGVP/01-annotate_variants_CAVATICA_input.R (hunk 6 above)
Comment (naqvia) [reply AutoGVP/01-annotate_variants_CAVATICA_input.R:90]: I have added this for readability purposes and to make debugging easier. At times to debug, we need to manually change and run the code in the script as opposed to on the command line. I prefer to have this for now especially as we finalize/modify the code.

**Comment by naqvia** [comment]
I added the gnomad filtering as optional without a default, and I also separated out the variant depth filter function. This way variants will be routinely removed if the variant depth is lower than the default (15). This modification should speed up the run-time greatly. I wonder if tested in larger cohorts, it'll suffice and ensure program doesn't crash/long time even without gnomad filter (cc @rjcorb). This way may allow us to report all the allele frequencies in INFO column and have the end user filter after the fact (ensuring reproducibility, universality, etc). I did need to change the default to 8 for the time being for testing purposes, bc none of the test file variants had above 15 read depth, so this should be changed back when testing in larger cohorts.

**Comment by jharenza** [comment]
@naqvia can you just update the read depth of your test files so this isn't forgotten later?

File: AutoGVP/01-annotate_variants_CAVATICA_input.R [hunk 7]
```diff
@@ -56,11 +56,11 @@ option_list <- list(
               help = "variant_summary file (format: variant_summary_2023-02.txt)"),
   make_option(c("--submission_summary"), type = "character",
               help = "specific submission summary file (format: submission_summary.txt)"),
-  make_option(c("--gnomad_variable"), type = "character",default = "Freq_gnomAD_genome_ALL",
+  make_option(c("--gnomad_variable"), type = "character",
```
Comment (jharenza) [inline AutoGVP/01-annotate_variants_CAVATICA_input.R:59]: can you update the name of this variable to `pop_variable` or `popfreq_variable` for population frequency? This can be gnomad, exac, or 1000g, or even a custom population, so we should not make it `gnomad`

**Comment by naqvia** [comment]
*note: this analysis may be obsolete with the bcftools solution.

**Comment by jharenza** [comment]
Agree - you can close this!
